### ISSUE TITLE
->* w/ opt and rest args is broken

### ISSUE THREAD

**Issue by pnwamk** [racket/typed-racket#614]
### What version of Racket are you using?
6.10.1.2
### What program did you run?
```racket
(: foo (->* () (String) #:rest Number Any))
(define foo
  (case-lambda [(str . nums)
                (string->number str) ;; TR error, expected String, given Number in str
                (+ 42 str)] ;; TR error, expected Number, given String in str
               [l 42]))
```
### What should have happened?
Optional arguments come _before_ rest arguments... right? I'm not sure exactly what we're doing in Typed Racket yet... but it seems wrong.

Here's a similar case-lambda in regular Racket with contracts that seems to work fine:

```racket
#lang racket/base
(require rackunit)


(module foo-module racket
  (provide/contract [foo (->* () (string?) #:rest (listof number?) any/c)])
  (define foo
    (case-lambda [(str . nums)
                  (apply + (string->number str) nums)]
                 [l 42])))

(require 'foo-module)

(check-equal? (foo) 42)
(check-equal? (foo "0") 0)
(check-equal? (foo "0" 1 2 3) 6)
(check-exn exn:fail? (λ () (apply foo "0"))) ;; expected a list? given "0"
(check-equal? (apply foo "0" (list 1 2 3)) 6)
```

This is likely related to previously reported issues involving optional and rest arguments (https://github.com/racket/typed-racket/issues/413 https://github.com/racket/typed-racket/issues/432).

Will need to dig deeper to see what the root cause is =\

**Comment by pnwamk** [comment]
working on a fix for this -- shouldn't be too complex I don't think...

**Comment by pnwamk** [comment]
Fixed by ffcf5afe921742abd768ce53ae80d3273dbf93e4.
